certbot-apache/certbot_apache/parser.py

```diff
@@ -163,15 +169,54 @@ def update_runtime_variables(self):
 
         self.variables = variables
 
-    def _get_runtime_cfg(self):  # pylint: disable=no-self-use
+    def update_includes(self):
+        """Get includes from httpd process, and add them to DOM if needed"""
+
+        # Find_dir iterates over configuration for Include and IncludeOptional
+        # directives to make sure we see the full include tree present in the
+        # configuration files
+        _ = self.find_dir("Include")
```

Comment:
Two questions here:
1. Do we also need to look for `IncludeOptional`? If not, why not?
2. Is it possible for the reverse problem you described to happen? That is, if the user's config includes `/path/to/file.conf`, will `DUMP_INCLUDES` ever include a wildcard like `/path/to/*.conf` in its output? This has the potential to break things for users where the Apache plugin currently works so I want to make sure we're not expecting breakage here.

Also, a nit. Personally, with the return value not being used I'd just change this line to `self.find_dir("Include")` and let Python automatically ignore the return value. If you like this better for some reason, feel free to leave it though.

Reply:
1) Yeah, actually `find_dir()` searches for both `Include` and `IncludeOptional` statements *in addition* to the provided search directive. So it would have worked perfectly with whatever argument, but I decided to use `"Include"` for readability.
2) No, `DUMP_INCLUDES` always returns the actual files that have been loaded.

Actually I'd prefer to leave it as it is for readability. I usually do the very thing you described, but decided to leave it in this case because the method is designed to actually return something, and it's used here in a bit hackish way against its intended purpose. We should refactor that functionality out of it at some point in the future.